torch/csrc/lazy/core/ir.h

```diff
@@ -88,8 +91,38 @@ class TORCH_API Node {
   // Contructor used to create leaf nodes.
   Node(OpKind op, size_t num_outputs, std::function<hash_t(bool)> node_hash_fn);
 
+  Node(OpKind op, OpList operands, std::vector<Shape>&& shapes,
```

Comment:
I wonder if we actually want all these constructors in Node base. I would prefer to clean it up (get rid of the SetShapeDeferred and shape_fn ones) based on unifying the remaining handwritten node classes to use 'compute_shape' functions.

Is it helpful for you to have these ctors for the migration period @JackCaoG?

Reply:
We don't need `shape_fn` and `SetShapeDeferred`. For now pytorch/xla still need to `shape_fn` but that is for the `xla::shape`. We have no use case of `shape_fn` for lazy::shape.

Reply:
@wconstab gentle reminder for this comment

Reply:
I think this needs more discussion with @JackCaoG to be totally sure.

For Torchscript backend, I think we don't actually need shapes at all :) but that also needs a little more discussion.  They are mainly there in TS backend as a way to prove complete coverage for once XLA starts using the shape infrastructure we promised to provide.

I'd say leave the ctors in for now though, unless they are a major burden, and we can possibly remove them later along with cleaning up who uses shape infra.